### PR TITLE
assimp: Add 5.1.6 and 5.2.2 + fix deps

### PR DESCRIPTION
**Pull request by intelligide**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by uilianries — DISMISSED** [review]

**Review by SSE4 — DISMISSED** [review]

**Review by StellaSmith — APPROVED** [review]

File: recipes/assimp/5.x/conanfile.py [hunk 1]
```diff
@@ -141,17 +149,18 @@ def _depends_on_zlib(self):
...
-        #   has 6.4.2, not API compatible with 4.8.8 vendored in assimp
         # - Open3DGC
-        # - openddlparser
         if tools.Version(self.version) < "5.1.0":
             self.requires("irrxml/1.2")
         else:
             self.requires("pugixml/1.11")
+            self.requires("openddl-parser/cci.20211217")
 
+        self.requires("zlib/1.2.11")
```
Comment (SpaceIm) [inline recipes/assimp/5.x/conanfile.py:159]:
```suggestion
        self.requires("zlib/1.2.11")
```

Not a direct dependency as far as I remember. It's a dependency of minizip

File: recipes/assimp/5.x/conanfile.py [hunk 2]
```diff
@@ -141,17 +149,18 @@ def _depends_on_zlib(self):
...
+        self.requires("stb/cci.20210910")
```
Comment (SpaceIm) [inline recipes/assimp/5.x/conanfile.py:160]:
```suggestion
```

not used by the library, only in samples

File: recipes/assimp/5.x/conanfile.py [hunk 3]
```diff
@@ -141,17 +149,18 @@ def _depends_on_zlib(self):
...
         # TODO: unvendor others libs:
-        # - clipper (required by IFC importer): can't be unvendored because CCI
-        #   has 6.4.2, not API compatible with 4.8.8 vendored in assimp
         # - Open3DGC
-        # - openddlparser
         if tools.Version(self.version) < "5.1.0":
             self.requires("irrxml/1.2")
         else:
             self.requires("pugixml/1.11")
+            self.requires("openddl-parser/cci.20211217")
```
Comment (SpaceIm) [inline recipes/assimp/5.x/conanfile.py:157]:
```suggestion
            self.requires("openddl-parser/cci.20211217")
            self.requires("stb/cci.20210910")
```

File: recipes/assimp/5.x/conanfile.py (hunk 2 above)
Comment (SpaceIm) [inline recipes/assimp/5.x/conanfile.py:160]:
```suggestion
```

**Review by SpaceIm — CHANGES_REQUESTED** [review]
Before 5.1.0, stb is used in samples only.

File: recipes/assimp/5.x/conanfile.py (hunk 1 above)
Comment (SpaceIm) [inline recipes/assimp/5.x/conanfile.py:159]:
```suggestion
```

zlib already required below depending on options.

File: recipes/assimp/5.x/conanfile.py [hunk 4]
```diff
@@ -187,45 +196,45 @@ def _patch_sources(self):
             tools.replace_in_file(os.path.join(
                 self._source_subfolder, "CMakeLists.txt"), before, after, strict=False)
         # Take care to not use these vendored libs
-        vendors = ["poly2tri", "rapidjson", "unzip", "utf8cpp", "zip"]
+        vendors = ["poly2tri", "rapidjson", "utf8cpp", "zip", "unzip", "stb", "zlib", "clipper"]
         if tools.Version(self.version) < "5.1.0":
             vendors.append("irrXML")
         else:
-            vendors.extend(["pugixml", "draco"])
+            vendors.extend(["pugixml", "draco", "openddlparser"])
```
Comment (SpaceIm) [inline recipes/assimp/5.x/conanfile.py:203]: Why openddlparser is not unvendored in assimp < 5.1.0 also?

Comment (intelligide) [reply recipes/assimp/5.x/conanfile.py:203]: Before 5.1.0, assimp used a modified and incompatible version of openddlparser

File: recipes/assimp/5.x/patches/0001-unvendor-deps-5.0.x.patch [hunk 5]
```diff
@@ -121,18 +142,39 @@
...
+ 				for(size_t a = 0; a < it->contour.size(); ++a)
+ 					worldSpaceContourVtx[a] = minv * IfcVector3(it->contour[a].x, it->contour[a].y, 0.0);
+-				
++
+ 				IfcVector3 contourNormal = TempMesh::ComputePolygonNormal(worldSpaceContourVtx.data(), worldSpaceContourVtx.size());
+-				
++
+ 				reverseCountourFaces = (contourNormal * basePolyNormal) > 0.0;
+ 			}
+
```
Comment (SpaceIm) [inline recipes/assimp/5.x/patches/0001-unvendor-deps-5.0.x.patch:174]:
```suggestion
```

File: recipes/assimp/5.x/patches/0001-unvendor-deps-5.0.x.patch [hunk 6]
```diff
@@ -9,9 +9,18 @@
...
  namespace Assimp {
+@@ -255,7 +255,7 @@
+             hexDiffuseColor.clear();
+             tmp.clear();
+             hexDiffuseColor = "#";
+-            
++
+             tmp = DecimalToHexa( color.r );
+             hexDiffuseColor += tmp;
+             tmp = DecimalToHexa( color.g );
```
Comment (SpaceIm) [inline recipes/assimp/5.x/patches/0001-unvendor-deps-5.0.x.patch:20]:
```suggestion
```

**Review by SpaceIm — COMMENTED** [review]
cleanup patches

File: recipes/assimp/5.x/conanfile.py [hunk 7]
```diff
@@ -141,14 +160,12 @@ def _depends_on_zlib(self):
...
         # TODO: unvendor others libs:
-        # - clipper (required by IFC importer): can't be unvendored because CCI
-        #   has 6.4.2, not API compatible with 4.8.8 vendored in assimp
         # - Open3DGC
-        # - openddlparser
         if tools.Version(self.version) < "5.1.0":
             self.requires("irrxml/1.2")
         else:
             self.requires("pugixml/1.11")
+            self.requires("openddl-parser/cci.20211217")
```
Comment (SpaceIm) [inline recipes/assimp/5.x/conanfile.py:168]: required by opengex importer only (`with_opengex=True`)

Comment (SpaceIm) [reply recipes/assimp/5.x/conanfile.py:168]: FYI, I've opened an issue long time ago to implement fine-grained dependencies handling in Assimp (and easier external dependencies usage): https://github.com/assimp/assimp/issues/3213

**Review by SpaceIm — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**